samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/assessment/AssessmentService.java

```diff
@@ -1118,6 +1118,191 @@ public SecurityAdvice isAllowed(String arg0, String arg1,
 			return text;
 		}
 
+
+	/**
+	 * 
+	 * @param assessment
+	 * @param bundle
+	 * @return
+	 */
+	public String exportAssessmentToMarkupText(AssessmentFacade assessment, Map<String,String> bundle) {
+		StringBuffer markupText = new StringBuffer(); 
+		int nQuestion = 1;
+		
+		Iterator<SectionFacade> sectionIter = assessment.getSectionArray().iterator();
+		while (sectionIter.hasNext()) {
+			SectionFacade section = sectionIter.next();
+			List<ItemDataIfc> items;
+			boolean hasRandomPartScore = false;
+			Double score = null;
+			boolean hasRandomPartDiscount = false;
+			Double discount = null;
+			
+			if (section.getSectionMetaDataByLabel(SectionDataIfc.AUTHOR_TYPE).equals(SectionDataIfc.QUESTIONS_AUTHORED_ONE_BY_ONE.toString()))
+  			{
+				items = section.getItemArray();
+  			}
+			else 
+			{
+				String requestedScore = (section.getSectionMetaDataByLabel(SectionDataIfc.POINT_VALUE_FOR_QUESTION) != null) ? 
+						                 section.getSectionMetaDataByLabel(SectionDataIfc.POINT_VALUE_FOR_QUESTION)	: "";
+						                 
+				if (requestedScore != null && !requestedScore.equals("")) {
+					hasRandomPartScore = true;
+					score = new Double(requestedScore);
+				}
+				
+				String requestedDiscount = (section.getSectionMetaDataByLabel(SectionDataIfc.DISCOUNT_VALUE_FOR_QUESTION) != null) ? 
+											section.getSectionMetaDataByLabel(SectionDataIfc.DISCOUNT_VALUE_FOR_QUESTION) : "";
+
+				if (requestedDiscount != null && !requestedDiscount.equals("")) {
+					hasRandomPartDiscount = true;
+					discount = new Double(requestedDiscount);
+				}
+				
+				QuestionPoolService qpService = new QuestionPoolService();
+				items = qpService.getAllItems(Long.valueOf(section.getSectionMetaDataByLabel(SectionDataIfc.POOLID_FOR_RANDOM_DRAW)));
+			}
+  				
+			for (ItemDataIfc item : items) 
+			{
+				// only exports these questions types
+				if (TypeIfc.MULTIPLE_CHOICE.intValue() != item.getTypeId() && TypeIfc.MULTIPLE_CORRECT.intValue() != item.getTypeId() 
+						&& TypeIfc.FILL_IN_BLANK.intValue() != item.getTypeId() && TypeIfc.FILL_IN_NUMERIC.intValue() != item.getTypeId()
+						&& TypeIfc.TRUE_FALSE.intValue() != item.getTypeId() && TypeIfc.ESSAY_QUESTION.intValue() != item.getTypeId()) {
+					continue;
+				}
+				
+				markupText.append(nQuestion + ". ");
+				if (hasRandomPartScore) {
+					markupText.append("(" + score + " " + bundle.get("points") + ")");
```

Comment:
Use a chain of `.appends()` rather than doing String concatenation. Same with below.